library/std/src/fs.rs

```diff
@@ -366,6 +366,32 @@ impl File {
         OpenOptions::new().write(true).create(true).truncate(true).open(path.as_ref())
     }
 
+    /// Creates a new file in read-write mode.
+    ///
+    /// This function will create a file if it does not exist, or return an error if it does. This
+    /// way, if the call succeeds, the file returned is guaranteed to be new.
+    ///
```

Comment:
The docs should point out that this is just a shorthand for precisely `OpenOptions::new().read(true).write(true).create_new(true).open(path.as_ref())` and not any kind of special implementation.